src/core/sort.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * @param {{priority: number}} a - First object with priority property.
+ * @param {{priority: number}} b - Second object with priority property.
+ * @returns {number} A number indicating the relative position.
+ * @ignore
+ */
+export const cmpPriority = (a, b) => a.priority - b.priority;
```

Comment:
Why does `cmpPriority` need to be exported. Should it be internal to this module?

Reply:
I just like to keep things public for potential reuse, but technically it can be private (since its not being used elsewhere)

Reply:
I'll merge this and hide cmpPriority in separate PR.